### PR TITLE
Creating a gallery view

### PR DESCRIPTION
**Pull request by LostButton**
Contributing.md needs some work, but so far the base is running off parcel 
the view uses Fancybox and has several online images posted inside. 
Click once to select the photo to view, and click it again to bring it to full size.
You can also drag the photos around when selected.

The navbar will also follow when scrolling down. 
The username is used as a placeholder for now nothing there yet
the About page is not there yet 
#2 
[test website ](https://lostbutton.github.io/my-photohub/gallery)

### CODE REVIEW

**Review by humphd — CHANGES_REQUESTED** [review]
Thanks for getting us started on this!  I love that you did a demo site to show how it runs.

My first initial feedback:

1. fancybox doesn't seem to be maintained any more.  Looking at https://npmtrends.com/fancybox-vs-lightgallery-vs-lightgallery.js-vs-photoswipe-vs-swipebox it seems that https://photoswipe.com/ is the one that more people use, or https://github.com/sachinchoolur/lightGallery.  I think we should use one of these.
2. Add this library to the `dependencies` of `package.json` vs. asking the user to install it manually
3. We won't have accounts or search, so you can remove both of those from the header
4. Blue text on black background is hard to read
5. I'd probably rename `gallery.html` to `index.html` so it resolves better
6. Use a newer version of Parcel (2.x)
7. Can we get rid of the JS in `gallery.js` and make it 100% static HTML?

**Comment by LostButton** [comment]
Which version of fancybox is this site talking about? because I know versions 1, 2, and 3 are now discontinued, but nothing is ever said about version 4.

**Comment by humphd** [comment]
https://www.npmjs.com/package/@fancyapps/fancybox hasn't been updated in 4 years, I'd say it's dead.  Also, it's based on jQuery, which isn't a good choice in 2022.

**Comment by LostButton** [comment]
I just changed it to photoswide and parcel should be 2.0 and above now. Its now 100% html and file has been renamed to index.html

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Getting Started
+
+Install Node.js if you haven't already and run:
+`npm install`
+
+Install **Fancybox** via the command:
```
Comment (humphd) [inline CONTRIBUTING.md:6]: This file is out of date now, please update to reflect the fact that you only need to do `npm install`

File: css/photoswipe.css [hunk 2]
```diff
@@ -0,0 +1,406 @@
+/*! PhotoSwipe main CSS by Dmytro Semenov | photoswipe.com */
```
Comment (humphd) [inline css/photoswipe.css:1]: We should be able to include this via npm from `node_modules` and have Parcel bundle it for us.

File: index.html [hunk 3]
```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width" />
+  
+    <title>My-PhotoHub📸</title>
+
+    <link rel="stylesheet" href="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/css/bootstrap.min.css"
```
Comment (humphd) [inline index.html:9]: Can we use Bootstrap from npm please?  Also, this is an old version of Bootstrap.

File: index.html [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    integrity="sha384-ggOyR0iXCbMQv3Xipma34MD+dH/1fQ784/j6cY/iJTQUOhcWr7x9JvoRxT2MZw1T" crossorigin="anonymous">
+
+    <link href="https://fonts.googleapis.com/css?family=Poppins" rel="stylesheet">
```
Comment (humphd) [inline index.html:12]: I think you can install this with npm and use Parcel to bundle

File: index.html [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+    
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/lightgallery-js/1.4.0/css/lightgallery.min.css" integrity="sha512-kwJUhJJaTDzGp6VTPBbMQWBFUof6+pv0SM3s8fo+E6XnPmVmtfwENK0vHYup3tsYnqHgRDoBDTJWoq7rnQw2+g==" crossorigin="anonymous" referrerpolicy="no-referrer" />
```
Comment (humphd) [inline index.html:14]: Remove this, we aren't using it now

File: index.html [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    
+    <script src="https://cdnjs.cloudflare.com/ajax/libs/lightgallery-js/1.4.0/js/lightgallery.min.js" integrity="sha512-b4rL1m5b76KrUhDkj2Vf14Y0l1NtbiNXwV+SzOzLGv6Tz1roJHa70yr8RmTUswrauu2Wgb/xBJPR8v80pQYKtQ==" crossorigin="anonymous" referrerpolicy="no-referrer">
```
Comment (humphd) [inline index.html:16]: Same

File: index.html [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+    </script>
+
+    
+<script type="module">
```
Comment (humphd) [inline index.html:20]: Move all this JS to an `app.js` or `index.js` file, and let Parcel do the work.

File: index.html [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+  import PhotoSwipeLightbox from 'https://unpkg.com/photoswipe/dist/photoswipe-lightbox.esm.js';
```
Comment (humphd) [inline index.html:21]: Pull these from npm/node_modules, not by URL.  Let Parcel bundle

File: index.html [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+  <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+    <a class="navbar-brand" href="#">My-PhotoHub📸</a>
+
+    <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarSupportedContent"
+        aria-controls="navbarSupportedContent" aria-expanded="false" aria-label="Toggle navigation">
+        <span class="navbar-toggler-icon"></span>
+    </button>
+    <div class="collapse navbar-collapse" id="navbarSupportedContent">
+        <ul class="navbar-nav mr-auto">
+            <li class="nav-item active">
```
Comment (humphd) [inline index.html:58]: What are the Home and About links for?  If you don't have content for them, I'd remove and add them only when we know what they are for.

File: package.json [hunk 10]
```diff
@@ -0,0 +1,24 @@
+{
+  "name": "My-PhotoHub",
+  "version": "1.0.0",
+  "description": "JavaScript example starter project",
```
Comment (humphd) [inline package.json:4]: Fix this to be an actual description of what we're building.

File: package.json [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+  "main": "index.html",
+  "scripts": {
+    "start": "parcel index.html --open",
+    "build": "parcel build index.html"
+  },
+  "dependencies": {
+    "parcel-bundler": "^2.0.0",
```
Comment (humphd) [inline package.json:11]: Change this to `parcel` and use `^2.7.0` to reflect current version https://www.npmjs.com/package/parcel

File: package.json [hunk 12]
```diff
@@ -0,0 +1,24 @@
...
+    "photoswipe": "^5.3.3"
+  },
+  "devDependencies": {
```
Comment (humphd) [inline package.json:14]: I don't believe this is necessary

**Review by humphd — CHANGES_REQUESTED** [review]

**Comment by humphd** [comment]
@LostButton do you want to wait to `push` until you're done?  I'm getting a lot of notifications on this PR.

**Comment by LostButton** [comment]
> @LostButton do you want to wait to `push` until you're done? I'm getting a lot of notifications on this PR.

the thing is its working perfectly when running locally, I'm having a hard time running it on github pages

**Comment by LostButton** [comment]
> What does "running it on GitHub Pages" mean? You don't run code on GitHub Pages.

it should be working on github pages as intended now

File: gallery.css [hunk 13]
```diff
@@ -0,0 +1,74 @@
...
+position: fixed;
+top: 0;
+left: 0;
+right: 0;
+z-index: 9999;
+}
+
+.navbar .navbar-brand{
+color: #FFF !important;
+
```
Comment (SerpentBytes) [inline gallery.css:36]: The project name could perhaps use a `margin-left` property.

File: index.html [hunk 14]
```diff
@@ -0,0 +1,59 @@
...
+
+  <body>
+
+  <!-- NAVIGATION BAR STARTS NAVIGATION BAR STARTS -->
+ 
+  <nav class="navbar navbar-expand-lg navbar-dark bg-dark">
+    <a class="navbar-brand" href="#">My-PhotoHub📸</a>
+</nav>
+
+    <p class="main">
```
Comment (SerpentBytes) [inline index.html:23]: Is it possible to render the content of the `p` tag, i.e. the gallery itself center-aligned horizontally using bootstrap classes and/or custom CSS styles? Or use some other technique to fix the inconsistencies in the left and right margins.

File: index.html [hunk 15]
```diff
@@ -0,0 +1,59 @@
...
+      <div id="gallery" class="gallery" itemscope itemtype="http://schema.org/ImageGallery">
+
```
Comment (SerpentBytes) [inline index.html:25]: Is it possible to render the content of the p tag, i.e. the gallery itself center-aligned horizontally using bootstrap classes and/or custom CSS styles? Or use some other technique to fix the inconsistencies in the left and right margins.

**Review by SerpentBytes — COMMENTED** [review]
I left some comments on your PR.

When I click on "test sample," it redirects me to [https://lostbutton.github.io/my-photohub/gallery](https://lostbutton.github.io/my-photohub/gallery), which renders a 404 error. 

When I navigate to https://lostbutton.github.io/my-photohub, it seems to work fine. Maybe you are deploying the website to the `root` path instead of `/gallery`?

**Comment by LostButton** [comment]
> I left some comments on your PR.
> 
> When I click on "test sample," it redirects me to https://lostbutton.github.io/my-photohub/gallery, which renders a 404 error.
> 
> When I navigate to https://lostbutton.github.io/my-photohub, it seems to work fine. Maybe you are deploying the website to the `root` path instead of `/gallery`?

No that's not the case, I commented previously that when it was changed to index that https://lostbutton.github.io/my-photohub would be the link its deployed to. But I will look into your suggestions

**Comment by SerpentBytes** [comment]
Lastly, on mobile devices tapping on a single image, shows two in preview. See below: 
image

File: gallery.css (hunk 13 above)
Comment (LostButton) [reply gallery.css:36]: Please explain, are you saying the logo should be on the left? or just padding so it doesn't scrape the side of the screen?

Comment (SerpentBytes) [reply gallery.css:36]: There should be some space before the project name.

File: gallery.css [hunk 16]
```diff
@@ -33,7 +32,6 @@ z-index: 9999;
 
 .navbar .navbar-brand{
 color: #FFF !important;
-
```
Comment (SerpentBytes) [inline gallery.css]: Hey, LGTM just add this on line 36: `margin-top: 15px`

**Review by SerpentBytes — APPROVED** [review]
LGTM.

**Comment by LostButton** [comment]
> Thanks for getting us started on this! I love that you did a demo site to show how it runs.
> 
> My first initial feedback:
> 
> 1. fancybox doesn't seem to be maintained any more.  Looking at https://npmtrends.com/fancybox-vs-lightgallery-vs-lightgallery.js-vs-photoswipe-vs-swipebox it seems that https://photoswipe.com/ is the one that more people use, or https://github.com/sachinchoolur/lightGallery.  I think we should use one of these.
> 2. Add this library to the `dependencies` of `package.json` vs. asking the user to install it manually
> 3. We won't have accounts or search, so you can remove both of those from the header
> 4. Blue text on black background is hard to read
> 5. I'd probably rename `gallery.html` to `index.html` so it resolves better
> 6. Use a newer version of Parcel (2.x)
> 7. Can we get rid of the JS in `gallery.js` and make it 100% static HTML?

Prof, for number 7 what do you mean get rid of js to make it 100% static HTML?

File: package.json [hunk 17]
```diff
@@ -0,0 +1,29 @@
+{
+  "name": "My-PhotoHub",
```
Comment (humphd) [inline package.json:2]: We have 3 separate PRs that are all doing the same thing (creating `package.json`).  A better approach for doing this in the future would be to start with a very small PR that adds a `pacakge.json` and then build on that.  Open source should happen in small bite-sized chunks.

Comment (LostButton) [reply package.json:2]: So what should I do with the package.json?

Comment (humphd) [reply package.json:2]: Moved the discussion to Slack

**Comment by LostButton** [comment]
> > Prof, for number 7 what do you mean get rid of js to make it 100% static HTML? I know its index now but the removing of JS still confuses me.
> 
> Ideally we should produce a static HTML page that has everything rendered in our build step vs. having to build the DOM dynamically on load. It will be way faster if we do.

For number 7, I can't remove the JS because the photoswipe relies on the modules inside. Unless you want me to write the script within the HTML file?

**Comment by humphd** [comment]
I don't mean remove all JavaScript.  I mean let's do as much in HTML as possible vs. having to set everything up at runtime with JS.  It's fine to augment that with JS afterward.

**Comment by LostButton** [comment]
> I don't mean remove all JavaScript.  I mean let's do as much in HTML as possible vs. having to set everything up at runtime with JS.  It's fine to augment that with JS afterward.

So would a better option would be to script what i have within the html file?

**Pull request merged**